File: node/temporary_api_key.js
const { SpeechClient, timestampToDate } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
    const response = await speechClient.createTemporaryApiKey({
        usage_type: "transcribe_websocket",
        client_request_reference: "test_reference",
    });

    console.log(`API key: ${response.key}`);
    console.log(`Expires: ${timestampToDate(response.expires_datetime)}`);
})();

File: python/temporary_api_key.py
from soniox.speech_service import SpeechClient
from soniox.utils import timestamp_to_datetime


def main():
    with SpeechClient() as client:
        response = client.CreateTemporaryApiKey(
            usage_type="transcribe_websocket",
            client_request_reference="test_reference",
        )

        print(f"API key: {response.key}")
        print(f"Expires: {timestamp_to_datetime(response.expires_datetime)}")


if __name__ == "__main__":
    main()
